src/Workspaces/Core/Portable/Workspace/WorkspaceChangeEventArgs.cs

```diff
@@ -4,6 +4,15 @@
 
 namespace Microsoft.CodeAnalysis
 {
+    /// <summary>
+    /// The <see cref="EventArgs"/> describing any kind of workspace change.
+    /// </summary>
+    /// <remarks>
+    /// When linked files are edited, one document change event is fired per linked file. All of
+    /// these events contain the same OldSolution, and they all contain the same NewSolution. This
+    /// is so that we can trigger document change events on all affected documents without
+    /// reporting intermediate states in which the linked file contents do not match.
+    /// </remarks>
     public class WorkspaceChangeEventArgs : EventArgs
```

Comment:
It might be worth adding comments on OldSolution/NewSolution to say that in general, if multiple documents are being changed at once you might see multiple events with same Old/New solution. Even outside the linked file case, I guess you could imagine that if I did a refactoring that touches two files we might start to consider raising that pattern too.